src/main/java/ch/njol/skript/FeatureConfig.java

```diff
@@ -0,0 +1,144 @@
+ */
+package ch.njol.skript;
+
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.zip.ZipEntry;
+import java.util.zip.ZipFile;
+
+import ch.njol.skript.config.Config;
+import ch.njol.skript.config.Node;
+import ch.njol.skript.config.OptionSection;
+import ch.njol.skript.config.SectionNode;
+import ch.njol.skript.util.FileUtils;
+import ch.njol.util.Pair;
+
+public class FeatureConfig {
+
+	public final static OptionSection alterSyntax = new OptionSection("alterSyntax");
+	public final static OptionSection features = new OptionSection("features");
+
+	private static Map<String, String> alteredPatterns = new HashMap<>();
+	private static List<String> disabledClassNames = new ArrayList<>();
+	private static List<String> disabledPatterns = new ArrayList<>();
+
+	private static boolean loaded;
+
+	public static Pair<Boolean, String[]> contains(String className, String... patterns) {
+		if (disabledClassNames.contains(className)) {
+			Skript.debug("Disabling feature '" + className + "' through the Features.sk config.");
+			return new Pair<Boolean, String[]>(true, patterns);
+		}
+
+		for (int i = 0 ; i < patterns.length ; i++) {
+			if (alteredPatterns.containsKey(patterns[i])) {
+				Skript.debug("Altering the pattern '" + patterns[i] + "' to '" + alteredPatterns.get(patterns[i]) + "'.");
+				patterns[i] = alteredPatterns.get(patterns[i]);
+				return new Pair<Boolean, String[]>(false, patterns);
+			} else if (disabledPatterns.contains(patterns[i])) {
+				Skript.debug("Disabling the feature '" + className + "' which had the exact pattern: '" + patterns[i] + "'.");
+				return new Pair<Boolean, String[]>(true, patterns);
+			}
+		}
+		return new Pair<Boolean, String[]>(false, patterns);
+	}
+
+	public static void load(File file) {
+		if (loaded)
+			return;
+		loaded = true;
+		File featureFile = new File(Skript.getInstance().getDataFolder(), "features.sk");
+		Config mc = null;
+		if (featureFile.exists()) {
+			try {
+				mc = new Config(featureFile, false, false, ":");
+			} catch (IOException e) {
+				e.printStackTrace();
+			}
+		} else {
+			try (ZipFile zip = new ZipFile(file)) {
+				File saveTo = null;
+				ZipEntry entry = zip.getEntry("features.sk");
+				if (entry != null) {
+					File af = new File(Skript.getInstance().getDataFolder(), entry.getName());
+					if (!af.exists())
+						saveTo = af;
+				} if (saveTo != null) {
+					try (InputStream in = zip.getInputStream(entry)) {
+						FileUtils.save(in, saveTo);
+					}
+				}
+			} catch (IOException e) {
+				if (Skript.debug())
+					Skript.exception(e);
+			} finally {
+				featureFile = new File(Skript.getInstance().getDataFolder(), "features.sk");
+				try {
+					mc = new Config(featureFile, false, false, ":");
+				} catch (IOException e) {
+					e.printStackTrace();
+				}
```

Comment:
We could probably avoid this duplication by setting `loaded` later and calling this method again after the file is created (assuming no exception occurred).